src/main/java/com/minecolonies/coremod/MineColonies.java

```diff
@@ -124,6 +125,10 @@ public MineColonies()
 
         InteractionValidatorInitializer.init();
         proxy.setupApi();
+
+        StructureName.HUTS.add("simplequarry");
+        StructureName.HUTS.add("mediumquarry");
+        StructureName.HUTS.add("largequarry");
```

Comment:
Should be consts somewhere.  Although why do these need explicit registration while the standard huts don't?  (Presumably the answer is "because it doesn't follow the `blockhut` naming convention -- should it?)

Should it just be registering everything in the building registry?  (Though this probably can't be done here, it would have to wait for common setup.  Bonus: this would also register huts from addon mods.)

Reply:
I was actually thinking about that, I think that `StructureName.HUTS` should be an list of `ResourceLocation` instead of `String`. In the meantime. In `com.minecolonies.apiimp.CommonMinecoloniesAPIImpl.java` we could add the lines VV to the buildingRegistry `RegistryBuilder`.
```java
                             .onAdd((owner, stage, id, buildingEntry, oldBuildingEntry) -> {
                                 StructureName.HUTS.add(buildingEntry.getBuildingBlock().getHutName());
                             })
```
(Note: Thought of this off hand with minimal research. I don't know how well this will work.)

Reply:
I'll adjust it to resource location in my open PR

Reply:
nvm, we don't have a resourcelocation input anyway, so it doesn't make any sense.

Reply:
But I'm making a PR for this.

Reply:
Made a PR for this.

Reply:
https://github.com/ldtteam/Structurize/pull/479#issuecomment-1017770816